python/lsst/source/injection/combine.py

```diff
@@ -0,0 +1,323 @@
+        return catalog
+
+    # TODO: implement support for combining multi-component objects on DM-42433
+    def _combine_multi_component_objects(
+        self,
+        catalog: Table,
+        tractInfo,
+        match_radius: float,
+    ) -> Table:
+        # TODO: when creating an injection catalog, make a groupId column to
+        # accurately identify multi-component objects. Use that here instead
+        # matching.
+        with Matcher(catalog[self.config.col_ra], catalog[self.config.col_dec]) as m:
+            idx, selfMatchIndices, _, dists = m.query_self(
+                match_radius,
+                min_match=2,
+                return_indices=True,
+            )
+        # get the indices of matched groups of sources
+        matched_inds_groups = [list(i) for i in set(map(tuple, idx)) if len(i) != 0]
+        # combine matched sources
+        for inds in matched_inds_groups:
+            # create a new empty row
+            catalog.add_row()
+            # take the position of the brightest component
+            bright_index = inds[
+                np.where(catalog[inds][self.config.col_mag] == catalog[inds][self.config.col_mag].min())[0][0]
+            ]
+            catalog[-1][self.config.col_ra] = np.min(catalog[bright_index][self.config.col_ra])
+            catalog[-1][self.config.col_dec] = np.min(catalog[bright_index][self.config.col_dec])
+            # concatenate source types to show all components
+            source_type = "+".join(catalog[self.config.col_source_type][inds].value)
+            catalog[-1][self.config.col_source_type] = source_type
+            # combine magnitudes with astropy units
+            fluxes = [(mag * u.ABmag).to(u.nanojansky) for mag in catalog[inds][self.config.col_mag]]
+            catalog[-1][self.config.col_mag] = np.sum(fluxes).to(u.ABmag)
+        # remove leftover duplicate component rows
+        catalog.remove_rows([matched_inds_groups])
+        return catalog
+
+    def _make_multiband_catalog(
+        self,
+        bands: list,
+        catalog_dict: dict,
+        match_radius: float,
+    ) -> Table:
+        """Combine multiple band-specific catalogs into one multiband
+        catalog.
+        """
+        # load the first catalog then loop to add info for the other bands
+        multiband_catalog = catalog_dict[bands[0]]
+        multiband_catalog.rename_column(self.config.col_mag, f"{bands[0]}_mag")
+        for band in bands[1:]:
+            # make a masked column for the new band
+            multiband_catalog.add_column(np.ma.masked_all(len(multiband_catalog)), name=f"{band}_mag")
+            # match the input catalog for this band to the existing
+            # multiband catalog
+            catalog_next_band = catalog_dict[band]
+            catalog_next_band.rename_column(self.config.col_mag, f"{band}_mag")
+            with Matcher(multiband_catalog[self.config.col_ra], multiband_catalog[self.config.col_dec]) as m:
```

Comment:
Is this doing an ra/dec match between bands? Rather than having consistent ids between bands?